### PR TITLE
Update to `Window.showDirectoryPicker()` and `Window.showSaveFilePicker()` and `Window.showOpenFilePicker()`

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

Update to `Window.showDirectoryPicker()` and `Window.showSaveFilePicker()` and `Window.showOpenFilePicker()`

### Motivation

1. add `{{Optional_Inline}}` to some some optional parameters
2. add the missing `id` and `startIn` parameters for `Window.showSaveFilePicker()` and `Window.showOpenFilePicker()`
3. add the missing `SecurityError` exception, see https://wicg.github.io/file-system-access/#permissions - permission request algorithm

### Additional details

https://developer.mozilla.org/en-US/docs/Web/API/Window/showDirectoryPicker

https://developer.mozilla.org/en-US/docs/Web/API/window/showOpenFilePicker

https://developer.mozilla.org/en-US/docs/Web/API/Window/showSaveFilePicker

https://wicg.github.io/file-system-access/#idl-index

image

### CODE REVIEW

File: files/en-us/web/api/window/showdirectorypicker/index.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ showDirectoryPicker()
       - : By specifying an ID, the browser can remember different directories for different
         IDs. If the same ID is used for another picker, the picker opens in the same
         directory.
-    - `mode`
+    - `mode` {{optional_inline}}
       - : A string that defaults to `"read"` for read-only access or `"readwrite"` for read
         and write access to the directory.
     - `startIn`
```
Comment (wbamberg) [inline files/en-us/web/api/window/showdirectorypicker/index.md:36]: It looks like `id` and `startIn` are optional as well: https://wicg.github.io/file-system-access/#wellknowndirectory-determine-the-directory-the-picker-will-start-in (in all 3 methods)

File: files/en-us/web/api/window/showopenfilepicker/index.md [hunk 2]
```diff
@@ -26,21 +26,28 @@ showOpenFilePicker()
...
+        `"downloads"`, `"music"`, `"pictures"`, or `"videos"`) to open the dialog in.
     - `types`
 
       - : An {{jsxref('Array')}} of allowed file types to pick. Each
         item is an object with the following options:
 
-        - `description`
-          - : An optional description of the category of files types allowed.
+        - `description` {{Optional_Inline}}
+          - : An optional description of the category of files types allowed. Default to be an empty string.
```
Comment (wbamberg) [inline files/en-us/web/api/window/showopenfilepicker/index.md:50]:
```suggestion
          - : An optional description of the category of files types allowed. Defaults to an empty string.
```

File: files/en-us/web/api/window/showdirectorypicker/index.md [hunk 3]
```diff
@@ -43,9 +43,11 @@ A {{jsxref("Promise")}} whose fulfillment handler receives a {{domxref('FileSyst
 
 ### Exceptions
 
-- `AbortError`
+- `AbortError` {{domxref("DOMException")}}
   - : Thrown if the user dismisses the prompt without making a selection, or if the user
-    agent deems the selected content to be too sensitive or dangerous
+    agent deems the selected content to be too sensitive or dangerous.
+- `SecurityError` {{domxref("DOMException")}}
+  - : An SecurityError is thrown if it was blocked by Same-Origin policy or it was not called via a user interaction such as a button press.
```
Comment (wbamberg) [inline files/en-us/web/api/window/showdirectorypicker/index.md:50]:
```suggestion
  - : Thrown if the call was blocked by the [same-origin policy](/en-US/docs/Web/Security/Same-origin_policy) or it was not called via a user interaction such as a button press.
```

Comment (wbamberg) [inline files/en-us/web/api/window/showdirectorypicker/index.md:50]:
```suggestion
  - : Thrown if the call was blocked by the [same-origin policy](/en-US/docs/Web/Security/Same-origin_policy) or it was not called via a user interaction such as a button press.
```

File: files/en-us/web/api/window/showopenfilepicker/index.md [hunk 4]
```diff
@@ -51,9 +58,11 @@ A {{jsxref("Promise")}} whose fulfillment handler receives an {{jsxref('Array')}
 
 ### Exceptions
 
-- `AbortError`
+- `AbortError` {{domxref("DOMException")}}
   - : An AbortError is thrown if a user dismisses the prompt without making a selection or
     if a file selected is deemed too sensitive or dangerous to be exposed to the website.
+- `SecurityError` {{domxref("DOMException")}}
+  - : An SecurityError is thrown if it was blocked by Same-Origin policy or it was not called via a user interaction such as a button press.
```
Comment (wbamberg) [inline files/en-us/web/api/window/showopenfilepicker/index.md:65]:
```suggestion
  - : Thrown if the call was blocked by the [same-origin policy](/en-US/docs/Web/Security/Same-origin_policy) or it was not called via a user interaction such as a button press.
```

File: files/en-us/web/api/window/showsavefilepicker/index.md [hunk 5]
```diff
@@ -51,9 +58,11 @@ A {{jsxref("Promise")}} whose fulfillment handler receives a {{domxref('FileSyst
 
 ### Exceptions
 
-- `AbortError`
+- `AbortError` {{domxref("DOMException")}}
   - : Thrown if the user dismisses the file picker without selecting or inputting a file,
     or if the user agent deems any selected files too sensitive or dangerous.
+- `SecurityError` {{domxref("DOMException")}}
+  - : An SecurityError is thrown if it was blocked by Same-Origin policy or it was not called via a user interaction such as a button press.
```
Comment (wbamberg) [inline files/en-us/web/api/window/showsavefilepicker/index.md:65]:
```suggestion
  - : Thrown if the call was blocked by the [same-origin policy](/en-US/docs/Web/Security/Same-origin_policy) or it was not called via a user interaction such as a button press.
```

**Review by wbamberg — COMMENTED** [review]
Thank you @skyclouds2001 !

File: files/en-us/web/api/window/showsavefilepicker/index.md [hunk 6]
```diff
@@ -26,21 +26,28 @@ showSaveFilePicker()
...
+    - `id` {{Optional_Inline}}
+      - : By specifying an ID, the browser can remember different directories for different
+        IDs. If the same ID is used for another picker, the picker opens in the same
+        directory.
+    - `startIn` {{Optional_Inline}}
+      - : A `FileSystemHandle` or a well known directory (`"desktop"`, `"documents"`,
+        `"downloads"`, `"music"`, `"pictures"`, or `"videos"`) to open the dialog in.
+    - `suggestedName` {{Optional_Inline}}
       - : A {{jsxref('String')}}. The suggested file name.
     - `types`
```
Comment (skyclouds2001) [inline files/en-us/web/api/window/showsavefilepicker/index.md:43]:
```suggestion
    - `types` {{Optional_Inline}}
```

File: files/en-us/web/api/window/showopenfilepicker/index.md [hunk 7]
```diff
@@ -26,21 +26,28 @@ showOpenFilePicker()
...
+      - : By specifying an ID, the browser can remember different directories for different
+        IDs. If the same ID is used for another picker, the picker opens in the same
+        directory.
+    - `multiple` {{Optional_Inline}}
+      - : A boolean value that defaults to `false`. When
+        set to `true` multiple files may be selected.
+    - `startIn` {{Optional_Inline}}
+      - : A `FileSystemHandle` or a well known directory (`"desktop"`, `"documents"`,
+        `"downloads"`, `"music"`, `"pictures"`, or `"videos"`) to open the dialog in.
     - `types`
```
Comment (skyclouds2001) [inline files/en-us/web/api/window/showopenfilepicker/index.md:44]:
```suggestion
    - `types` {{Optional_Inline}}
```

**Review by wbamberg — APPROVED** [review]
👍 thank you @skyclouds2001 !

**Pull request merged**